SparkPost/2web2ui
# SA-1585: Saved Reports, Create a new Saved Report

### What Changed
 - Installed react-hook-form
 - Made a wrapper for Checkboxes for react-hook-form
 - Added action/reducer pair for create saved report
 - Added button to open up modal to save new report
 - Added modal to save new report

### How To Test
 - Make sure user ui option `allow_saved_reports` is enabled
 - Go to signals/analytics
 - Click on Save new Report button
 - Click on save report. Verify that the validation won't allow you to submit empty forms.
 - Fill out the text fields and click save report.
 - Verify that the success snackbar appears.
 - Verify that it actually saved. (Do this through Postman/CURL or a branch with the GET request enabled.)


### To Do
- [x] UX approval
